src/client/object.rs

```diff
@@ -289,13 +276,68 @@ impl<'a> ObjectClient<'a> {
             .headers(self.0.get_headers().await?)
             .send()
             .await?;
+
         if resp.status() == StatusCode::NOT_FOUND {
             Err(crate::Error::Other(resp.text().await?))
         } else {
-            Ok(resp.error_for_status()?.bytes().await?.to_vec())
+            Ok(resp.error_for_status()?)
         }
```

Comment:
I created a non-pub helper function `download_response` to unify the common download code paths. I also noticed checking for `StatusCode::NOT_FOUND` was only done for `download`, not `download_streamed`. I think it makes more sense to have that unified here, but it might be a breaking change? 

Up for you to decide, I'm happy to move that error handling around.

Usability note: Often the google error responses include helpful content that is being discarded here. I think on http status errors, we should actually try to convert it into an `Error::Google(GoogleErrorResponse)`. Maybe in a separate MR?